### PR TITLE
Fix some edge case bugs in dockerise

### PR DESCRIPTION
**Pull request by CMurtagh-LGTM**
Whilst looking at running our code base with podman I ran into a couple bugs that could possibly happen even when running with docker.

### CODE REVIEW

File: tools/utility/dockerise/platform.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def selected(repository):
 
         names = [
             tag.split(":")[-1]
-            for tag in img_info[0]["RepoTags"]
```
Comment (KipHamiltons) [inline tools/utility/dockerise/platform.py:28]: Double list comprehension makes my brain hurt. Could you please split the assignment of `names` into smaller blocks? I am having trouble _comprehending_ what this does at all. (I wouldn't class this as scope creep, because the replacement you have offered takes the complexity over a threshold where my brain hurts.)

File: tools/utility/dockerise/platform.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def selected(repository):
         else:
             print("WARNING There are multiple platforms with the same image tag.")
             print("        The possible tags are [{}]".format(", ".join(names)))
-            platform = list(sorted(names))[0]
```
Comment (KipHamiltons) [inline tools/utility/dockerise/platform.py:41]: Very curious what prompted this change

File: tools/utility/dockerise/run.py [hunk 3]
```diff
@@ -17,7 +17,11 @@
 
 def _is_docker():
     path = "/proc/self/cgroup"
-    return os.path.exists("/.dockerenv") or os.path.isfile(path) and any("docker" in line for line in open(path))
+    return (
+        os.path.exists("/.dockerenv")
+        or os.path.isfile(path)
```
Comment (KipHamiltons) [inline tools/utility/dockerise/run.py:22]: Can you please add brackets around this condition?

reading this, it's not clear to me when:
- the first condition is false
- the second condition is true

what happens? 

If operator precedence of `or` and `and` in python is equivalent, then the third condition never matters - the `and` requires the second condition to be true, but if the second condition is true and the precedence is the same, then the `or` is true, and the whole statement is true, because short-circuit.
Please just make this more clear, either way.

**Review by KipHamiltons — CHANGES_REQUESTED** [review]

File: tools/utility/dockerise/platform.py (hunk 2 above)
Comment (TrentHouliston) [reply tools/utility/dockerise/platform.py:41]: Likely an indexed access, what is returned by sorted is an iterator, not a list.
So previously `platform[0]` would have been invalid

File: tools/utility/dockerise/platform.py (hunk 1 above)
Comment (TrentHouliston) [reply tools/utility/dockerise/platform.py:28]: This is more of a filter than a double list comprehension.
But I agree with kip in this case, easier to add a new line that splits out your `/`

File: tools/utility/dockerise/platform.py (hunk 2 above)
Comment (KipHamiltons) [reply tools/utility/dockerise/platform.py:41]: [the python docs](https://docs.python.org/3/library/functions.html#sorted) say that it returns a new list though?
Based on that, it seems to me that this cast to a list is simply just redundant.

Comment (CMurtagh-LGTM) [reply tools/utility/dockerise/platform.py:41]: I was getting an error when I wrote it, now I cannot reproduce the error :slightly_frowning_face:. It is redundant code though, so I'm going to leave it in the pr.

**Review by KipHamiltons — COMMENTED** [review]
LGTM

**Review by KipHamiltons — APPROVED** [review]
